### PR TITLE
Added steps, @poltergeist tag for testing deleting membership apps, companies

### PR DESCRIPTION
**Pull request by weedySeaDragon**
part 1: Adds Cucumber steps so that PT #138063171 can be tested.   (and a little grammar fix to the steps)

(This is the first PR for the PT story below.)

PT Story: **Delete company associated with deleted membership application. (see details)**
#138063171 https://www.pivotaltracker.com/story/show/138063171

Changes proposed in this pull request:
1.  added steps to show how many companies or business categories are on a page:
   `I should see "n" companies`  (where `n` is a number) 
   `I should see "n" business categories`  (where `n` is a number) 
   both use specific css tags to identify what is a 'company' or 'business category' on a page
2. added steps to confirm or dismiss a popup like the one that asks a user to confirm that they want to delete something:
    `[When] I confirm popup`
    `[When] I dismiss popup`
    These **require** the poltergeist driver.  So any scenario using them must include either the `@javascript` tag or the new `@poltergeist` tag below.

3. added tag `@poltergeist` to signal that the poltergeist driver is required.  Since javascript isn't explicitly involved, I added this tag instead of using `@javascript`

4. Corrected the grammar in a step. It was `I am the page for company number "n"` it should be `I am on the page for company number "n"`
    The (correct) `on ` is optional so that the previous wrong-grammar steps are still just fine, but you can know write it correctly by including `on`.

@patmbolger @thesuss

### CODE REVIEW

**Comment by thesuss** [comment]
These changes are all in another PR already. So I close this.

File: features/step_definitions/membership_application_steps.rb
Comment (thesuss) [inline features/step_definitions/membership_application_steps.rb:5]: @weedySeaDragon this one is not represented in https://github.com/AgileVentures/shf-project/pull/128/files - everything else in this PR is. 
So either add this to the 128 branch or make a separate PR with this change?

File: features/support/hooks.rb
Comment (patmbolger) [inline features/support/hooks.rb:8]: These are duplicative of the hooks for ```@javascript```

**Comment by weedySeaDragon** [comment]
@thesuss 
I wanted to separate these changes from PR #128  (and #129) so that they'd be easier to review and so that I'd wouldn't have another PR with so many.

The other PRs do include these commits because the next PRs (128, 129) depend on this work.  Once this PR is merged, those duplicative commits will 'go away' in the other PRs.   

Doing it this way also means that I can continue development -- I don't have to wait for these small PRs to be reviewed and merged before I keep going. 

IOW, this is an r to break up a larger piece of work into separate PRs.  So -- based on the comments here, this may not be the best way to accomplish that. :-)

Perhaps I should have marked the other PRs (128, 129) as WIP and put a comment in there?

This is definitely something we should talk about in our next meeting(s). (without the client, of course)

**Comment by weedySeaDragon** [comment]
@patmbolger I had no idea that I could pass a list of tags to a hook in cucumber.  thanks for nudging me to learn about it! :-)

**Review by patmbolger — APPROVED** [review]
PR looks good to me.

**Comment by weedySeaDragon** [comment]
@patmbolger can you merge it, please?  it's bad form to merge your own PRs in shf-project

**Pull request merged**
